WIP: Cleaned up properties in Product structure of json - (Task #220)

PhilMFischer opened the pull request:
- Not exporting shape anymore
- Not exporting stl anymore
- renamed properties to be more consistent
- added testcase to check for sub assemblies and their part
representatives in the json export
- adjusted the imports accordingly

---
Task #220: Clean up mechanical export products

Commit:
Cleaned up properties in Product structure of json - (Task #220)

- Not exporting shape anymore
- Not exporting stl anymore
- renamed properties to be more consistent
- added testcase to check for sub assemblies and their part
representatives in the json export
- adjusted the imports accordingly

---
Task #220: Clean up mechanical export products

andriikovalov-dlr reviewed (commented):
This kind of makes sense to avoid ambiguity in where shape is defined. But what if we override shape in a Configuration (e.g. add an stl), then with this approach it will be ignored in the export, then if there are changes to the shape on CAD side they will be reimported into Definition, but Configuration won't change (since it overrides).
If this is intended, I'll approve it.

Commit:
Improving test cases - (Task #228)

- implementing all three cases for visualization CAs on sub systems
---
Task #228: Port Phase B Concepts

Commit:
Finalized and cleaned up test cases - (Task #227)

- Added protected method to set up data which is common to all three
test cases
- Added test case 1 for a tree with sub system without Visualization CA
- Added test case 2 for a tree with sub system with Visualization CA but
shape set to NONE
- Added test case 3 for a tree with sub system with Visualization CA and
shape set to a primitive

---
Task #227: Integrate the Phase B concepts into the build process

SaMuellerDLR commented:
Created new PR to replace this one https://github.com/virtualsatellite/VirtualSatellite4-Core/pull/322
